packages/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -228,9 +230,10 @@ const defaultColumnsStyles = {
 export const serializeColumn = (column: GridColDef, columnsStyles: ColumnsStylesInterface) => {
   const { field, type } = column;
 
+  const headerName = valueHelper(column.headerName);
   return {
     key: field,
-    headerText: column.headerName ?? column.field,
+    headerText: isString(headerName) ? headerName : column.field,
```

Comment:
You're using the pattern `isString(headerName) ? headerName : col.field` a lot, this should probably be a function on its own. I don't like the 2 new functions (`isString` & `value`), they feel extremely generic.